### PR TITLE
Add missing logic for DefaultConfigOverwrite

### PR DESCRIPTION
**Pull request by gibizer**
We only want to support the policy.yaml file passthrough for PlacementAPI

The kolla config files are modified to move the supported files in place.

Implements: [OSPRH-2503](https://issues.redhat.com//browse/OSPRH-2503)

### CODE REVIEW

File: api/v1beta1/placementapi_webhook.go [hunk 1]
```diff
@@ -97,3 +118,30 @@ func (r *PlacementAPI) ValidateDelete() error {
...
 	return nil
 }
+
+func (r PlacementAPISpec) ValidateCreate(basePath *field.Path) field.ErrorList {
+	return r.ValidateDefaultConfigOverwrite(basePath)
+}
+
+func (r PlacementAPISpec) ValidateUpdate(old PlacementAPISpec, basePath *field.Path) field.ErrorList {
+	return r.ValidateDefaultConfigOverwrite(basePath)
+}
```
Comment (gibizer) [inline api/v1beta1/placementapi_webhook.go:128]: openstack-operator needs to call this

Comment (gibizer) [reply api/v1beta1/placementapi_webhook.go:128]: Proposed in https://github.com/openstack-k8s-operators/openstack-operator/pull/660

**Review by mrkisaolamb — COMMENTED** [review]
It will be nice if we also clean comments about DefaultConfigOverwrite in https://github.com/openstack-k8s-operators/placement-operator/blob/main/controllers/placementapi_controller.go#L1177

**Comment by gibizer** [comment]
> It will be nice if we also clean comments about DefaultConfigOverwrite in https://github.com/openstack-k8s-operators/placement-operator/blob/main/controllers/placementapi_controller.go#L1177

Done.

**Review by mrkisaolamb — APPROVED** [review]
lgtm

**Comment by gibizer** [comment]
/test placement-operator-build-deploy-kuttl

infra operator failed to start
```
++ oc get csv -n openstack-operators -o 'jsonpath={range .items[*]}{@.metadata.name}{"\n"}{end}'
++ grep -E '^infra-operator\.v'
+ CSVNAME=
+ '[' -z '' ']'
+ echo NOTFOUND
+ exit 1
+ rc=124
+ popd
~/install_yamls
+ exit 124
make[1]: *** [Makefile:511: wait] Error 124
make[1]: Leaving directory '/alabama/install_yamls'
make: *** [Makefile:977: memcached_deploy] Error 2
```

**Pull request merged**